File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {

    List<List<Integer>> res = new ArrayList<>();
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        
        process(nums, 0, target, new ArrayList<Integer>());
        return res ;
    }

    void process(int[] nums, int index, int remain,  List<Integer> path){
        
         if(remain < 0 ) return ;
         if(remain == 0){
            res.add(new ArrayList<>(path)) ;
            return ;
         }

         for(int i = index ; i < nums.length ; i++){
            path.add(nums[i]);
            process(nums, i, remain - nums[i], path) ;
            path.remove(path.size() -1);
         }
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:

        l = 0 
        res = []
        for r in range(k-1, len(nums)):
            m = float('-inf')
            for i in nums[l:r+1]:
                m = max(m, i)
            res.append(m)

            l += 1
        return res  


        